crates/turborepo-lib/src/run/summary/task.rs

```diff
@@ -131,30 +131,36 @@ impl TaskCacheSummary {
     }
 }
 
-impl From<Option<CacheResponse>> for TaskCacheSummary {
-    fn from(value: Option<CacheResponse>) -> Self {
-        value.map_or_else(Self::cache_miss, |CacheResponse { source, time_saved }| {
-            let source = CacheSource::from(source);
-            // Assign these deprecated fields Local and Remote based on the information
-            // available in the itemStatus. Note that these fields are
-            // problematic, because an ItemStatus isn't always the composite
-            // of both local and remote caches. That means that an ItemStatus might say it
-            // was a local cache hit, and we return remote: false here. That's misleading
-            // because it does not mean that there is no remote cache hit,
-            // it _could_ mean that we never checked the remote cache. These
-            // fields are being deprecated for this reason.
-            let (local, remote) = match source {
-                CacheSource::Local => (true, false),
-                CacheSource::Remote => (false, true),
-            };
-            Self {
-                local,
-                remote,
-                status: CacheStatus::Hit,
-                source: Some(source),
-                time_saved,
+// This is an `Option<CacheResult<CacheHitMetadata>>` because we could fail
+// to fetch from cache (giving `None`), and we could also get a miss
+// `Some(CacheResult::Miss)`
```

Comment:
I think it might make life easier to just keep this as `Option<CacheHitMetadata>` since we treat `None` and `Some(None`) the same.